enhancements/installer/kubevirt-ipi.md

```diff
@@ -0,0 +1,346 @@
+---
+title: KubeVirt-platform-provider
+authors:
+  - "@ravidbro"
+reviewers:
+
+approvers:
+
+creation-date: 2020-07-14
+last-updated: 2020-07-14
+status: implementable
+---
+
+# KubeVirt-platform-provider
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [OpenShift/docs]
+
+## Open Questions [optional]
+
+## Summary
+
+This document describes how [KubeVirt][kubevirt-website] becomes an infra provider for OpenShift.
+
+`KubeVirt` is a virtualization platform running as an extension of Kubernetes.
+
+ We want to create a tenant cluster on top of existing OpenShift/Kubernetes cluster by creating
+ virtual machines by KubeVirt for every node in the tenant cluster (master and workers nodes)
+ and other OpenShift/Kubernetes resources to allow **users** (not admins) of the infra cluster
+ to create a tenant cluster as it was an application running on the infra cluster.
+ We will implement all the components needed for the installer and cluster-api-provider
+ for the machine-api to allow post-install operations of resizing the cluster.
+
+
+## Motivation
+
+- Achieve true multi-tenancy of OpenShift were each tenant has dedicated control plane 
+ and has full control on its configuration to allow each user to install different versions
+ with a different configuration as permissions settings and installed operators. 
+
+### Goals
+
+- provide a way to install OpenShift on KubeVirt infrastructure using 
+  the installer - an IPI installation. (install-time)
+- implementing a cluster-api provider to provide scaling and managing the cluster
+  nodes (used by IPI, and useful for UPI, and node management/fencing) (post-install)
+- Provide multi-tenancy and isolation between the tenant clusters.
+- Provide tenant clusters with different versions and different configuration as permissions settings
+and installed operators.
+
+### Non-Goals
+- Implement UPI flow.
+
+## Proposal
+
+This provider enables the OpenShift Installer to provision VM resources in 
+KubeVirt infrastructure, that will be used as worker and masters of the clusters. It 
+will also create the bootstrap machine, and the configuration needed to get
+the initial cluster running by supplying a DNS service and load balancing.
+
+We want to approach deployment on KubeVirt as deployment on cloud similar to the
+deployments we have on public clouds as AWS and GCP rather than virtualization platform in a way 
+that the machine's network will be private, and the relevant endpoints will be exposed out of the
+cluster with platform services as we can or pods deployed in the infrastructure cluster to supply the services
+as DNS and load balancing.
+
+We see two main network options for deployment over KubeVirt:
```

Comment:
Will we be able to support cluster using third-party network SDNs like Calico or similar?

Reply:
We will be able to support only CNIs that are supported by CNV,
Right now, the only option that supported is Multus with bridge CNI,
By theory, every Multus CNI that supplies DHCP and routes in and out should work too.